File: artwork-service/src/main/java/com/gallery/artworkservice/domain/ArtworkID.java
package com.gallery.artworkservice.domain;

import java.util.Objects;

public class ArtworkID {
    private int id;

    public ArtworkID() {
        this.id = 0;
    }

    public ArtworkID(int id) {
        this.id = id;
    }

    public ArtworkID(ArtworkID other) {
        this.id = other.id;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ArtworkID)) return false;
        ArtworkID that = (ArtworkID) o;
        return id == that.id;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: app/ArtworkService/Domain/ArtworkID.cs
namespace ArtworkService.Domain
{
    public class ArtworkID
    {
        private int _artworkId;

        public ArtworkID()
        {
            this._artworkId = 0;
        }

        public ArtworkID(int artworkId)
        {
            this._artworkId = artworkId;
        }

        public ArtworkID(ArtworkID artworkID)
        {
            this._artworkId = artworkID._artworkId;
        }

        public int ArtworkId
        {
            get { return this._artworkId; }
            set { this._artworkId = value; }
        }
    }
}
